esmlab/regrid.py

```diff
@@ -0,0 +1,243 @@
+import logging
+import os
+
+import ESMF
+import numpy as np
+import xarray as xr
+import xesmf as xe
+
+from .config import SETTINGS
+
+logger = logging.getLogger(__name__)
+
+
+class _grid_ref(object):
+    """Get ESMF grid object from grid file.
+    """
+
+    def __init__(self, name, overwrite_existing=False):
+        self.name = name
+        grid_file_dir = SETTINGS["gridfile_directory"]
+        self.scrip_grid_file = f"{grid_file_dir}/{self.name}.nc"
+
+        self._gen_grid_file(overwrite_existing=overwrite_existing)
+
+        self._esmf_grid_from_scrip()
+
+    def _gen_grid_file(self, overwrite_existing):
+        """Generate a grid file.
+        """
+        if os.path.exists(self.scrip_grid_file) and not overwrite_existing:
+            return
+
+        raise NotImplementedError("grid file generation is not implemented")
+
+    def _esmf_grid_from_scrip(self):
+        """Generate an ESMF grid object from a SCRIP grid file.
+        """
+
+        if not os.path.exists(self.scrip_grid_file):
+            raise ValueError(f"file not found: {self.scrip_grid_file}")
+
+        self.ds = xr.open_dataset(self.scrip_grid_file)
+
+        self.grid = ESMF.Grid(
+            filename=self.scrip_grid_file,
+            filetype=ESMF.api.constants.FileFormat.SCRIP,
+            add_corner_stagger=True,
+        )
+
+        self.shape = self.ds.grid_dims.values
+        mask = self.grid.add_item(ESMF.GridItem.MASK)
+        mask[:] = self.ds.grid_imask.values.reshape(self.shape[::-1]).T
+
+
+class regridder(object):
+    """Class to enable regridding between named grids.
+    """
+
+    def __init__(
+        self, name_grid_src, name_grid_dst, method="bilinear", overwrite_existing=False
+    ):
+        """
+        Parameters
+        ----------
+        name_grid_src : string
+                      Name of source grid.
+        name_grid_dst : string
+                      Name of destination grid.
+        method : string, optional
+                Regridding method. Options are:
+
+                - 'bilinear'
+                - 'conservative'
+                - 'patch'
+                - 'nearest_s2d'
+                - 'nearest_d2s'
+        overwrite_existing : logical, optional [Default=False]
```

Comment:
In the doc-string,  can you change `overwriting_existing` type from `logical` to `bool`